packages/shared/lib/core/profile/stores/active-profile.store.ts

```diff
@@ -22,23 +22,35 @@ export function updateActiveProfileSettings(payload: Partial<IProfileSettings>):
     }))
 }
 
-export function addAccountMetadataToActiveProfile(metadata: IAccountMetadata): void {
-    activeProfile?.update((state) => ({
-        ...state,
-        accountMetadata: [...(state?.accountMetadata ?? []), metadata],
-    }))
+export function addAccountPersistedDataToActiveProfile(
+    accountIndex: number,
+    accountPersistedData: IAccountPersistedData
+): void {
+    activeProfile?.update((state) => {
+        if (!state?.accountPersistedData) {
+            state.accountPersistedData = {}
+        }
+        state.accountPersistedData[accountIndex] = accountPersistedData
+        return state
+    })
 }
 
-export function getAccountMetadataByIndex(index: number): IAccountMetadata {
-    const { accountMetadata } = get(activeProfile)
-    return accountMetadata.find((metadata) => metadata.index === index)
+export function getActiveProfilePersistedAccountData(accountIndex: number): IAccountPersistedData | undefined {
+    return get(activeProfile)?.accountPersistedData?.[accountIndex]
 }
 
-export function updateAccountMetadataOnActiveProfile(index: number, metadata: Partial<IAccountMetadata>): void {
-    activeProfile?.update((state) => ({
-        ...state,
-        accountMetadata: state?.accountMetadata.map((existingValue) =>
-            existingValue.index === index ? { ...existingValue, ...metadata } : existingValue
-        ),
-    }))
+export function updateAccountPersistedDataOnActiveProfile(
+    accountIndex: number,
+    partialAccountPersistedData: Partial<IAccountPersistedData>
+): void {
+    activeProfile?.update((state) => {
+        if (!state?.accountPersistedData) {
+            state.accountPersistedData = {}
```

Comment:
```suggestion
            state.accountPersistedData = []
```

Reply:
The persisted profile looks like

```
export interface IPersistedProfile {
    id: string
    name: string
    type: ProfileType
    network: IPersistedNetwork
    lastStrongholdBackupTime: Date
    settings: IProfileSettings
    accountPersistedData: {
        [accountId: string]: IPersistedAccountData
    }
    isDeveloperProfile: boolean
    hasVisitedDashboard?: boolean
    lastUsedAccountIndex?: number
    clientOptions: IClientOptions
    forceAssetRefresh: boolean
    strongholdVersion: StrongholdVersion
    pfp?: INft
}
```

hence `accountPersistedData` should be initialized as `{}`

Reply:
ah yes, i just saw that you updated this :+1: